### PR TITLE
Task2

### PR DESCRIPTION
**Pull request by uzairpm**
Pull request for my second task.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
-  "scripts": {
-    "start": "react-scripts start",
-    "build": "react-scripts build",
-    "test": "react-scripts test --env=jsdom",
-    "eject": "react-scripts eject"
-  }
-}
\ No newline at end of file
+{
+    "name": "react_task2",
```
Comment (thebergamo) [inline package.json]: Keep a unique name in package.json, to make clear for what tasks this changes are, is better keep it in the branch name.

File: package.json [hunk 2]
```diff
@@ -1,16 +1,44 @@
...
+    "version": "1.0.0",
+    "description": "App built as task 2 for the react mentoring program",
+    "main": "index.js",
+    "scripts": {
+        "build": "webpack --config ./webpack.config.js --mode production --env.production",
```
Comment (thebergamo) [inline package.json]: You can create some smaller npm scripts and reuse theirs to do not repeat same commands with different flags all the time

File: dist/index.html [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
```
Comment (thebergamo) [inline dist/index.html]: Put directory `dist` in .gitignore

File: webpack.config.dev.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+var path = require('path');
+var webpack = require('webpack');
+// const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
```
Comment (thebergamo) [inline webpack.config.dev.js]: Avoid to leave code that is commented. You can create a conditional execution of this code by env variables if necessary

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            PRODUCTION: JSON.stringify(true),
+            VERSION: JSON.stringify('1.0'),
+            'process.env.NODE_ENV': JSON.stringify('"production"')
+        }),
+        new UglifyJSPlugin()
+        /* new CompressionPlugin({
+            algorithm: 'gzip'
+        }) */
+    ],
+    devServer: {
```
Comment (thebergamo) [inline webpack.config.js]: Don't think we need to have a devServer in production build :D

File: package.json [hunk 6]
```diff
@@ -1,16 +1,44 @@
...
+        "start:dev": "webpack-dev-server --config ./webpack.config.dev.js --mode development --env.development",
+        "start": "webpack-dev-server --config ./webpack.config.js --mode production --env.production",
```
Comment (thebergamo) [inline package.json]: As you are already defining this production value inside your "env" variable, I don't think it's really necessary here

File: dist/index.html (hunk 3 above)
Comment (uzairpm) [reply dist/index.html]: Okay.

File: package.json (hunk 1 above)
Comment (uzairpm) [reply package.json]: Okay

File: webpack.config.dev.js (hunk 4 above)
Comment (uzairpm) [reply webpack.config.dev.js]: Okay.

File: package.json (hunk 2 above)
Comment (uzairpm) [reply package.json]: Done.

File: package.json (hunk 6 above)
Comment (uzairpm) [reply package.json]: okay

**Pull request merged**